File: redis-example/src/main/java/com/example/redisexample/controller/TestController.java
package com.example.redisexample.controller;


import com.example.redisexample.domain.Point;
import com.example.redisexample.repository.PointRedisRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

@RestController
@RequiredArgsConstructor
public class TestController {

    private final PointRedisRepository pointRedisRepository;

    @GetMapping("/point")
    public Point visit(String id) {
        Optional<Point> optional = pointRedisRepository.findById(id);
        if (optional.isPresent()) {
            Point point = optional.get();
            return pointRedisRepository.save(point);
        }

        return pointRedisRepository.save(Point.builder()
                .id(id)
                .amount(123L)
                .refreshTime(LocalDateTime.now())
                .build());
    }

    @GetMapping("/list")
    public List<Point> getAll() {
        return (List<Point>) pointRedisRepository.findAll();
    }
}

File: mongo-decimel-example/src/main/kotlin/org/example/mongodecimelexample/controller/TestController.kt
package org.example.mongodecimelexample.controller

import org.example.mongodecimelexample.mongo.TestDataMongo
import org.example.mongodecimelexample.mongo.TestDataMongoRepository
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import java.math.BigDecimal

@RestController
class TestController(
    private val testDataMongoRepository: TestDataMongoRepository,
) {

    @GetMapping("/save")
    fun saveData(@RequestParam price: BigDecimal) {
        testDataMongoRepository.save(
            TestDataMongo(
                price = price
            )
        )
    }

    @GetMapping("/gt-price")
    fun findGtPrice(@RequestParam price: BigDecimal) = testDataMongoRepository.findByGtPrice(price)
}